src/main/java/mops/businesslogic/directory/DirectoryServiceImpl.java

```diff
@@ -183,9 +184,18 @@ public Directory deleteFolder(Account account, long dirId) throws MopsException
         securityService.checkReadPermission(account, directory);
         List<FileInfo> fileInfos = fileInfoService.fetchAllFilesInDirectory(dirId);
 
-        return fileInfos.stream() //this is a stream not violation of demeter's law
+        List<FileInfo> results = fileInfos.stream() //this is a stream not violation of demeter's law
                 .filter(query::checkMatch)
                 .collect(Collectors.toList());
+
+        List<FileInfo> subResults = new ArrayList<>();
+        for (Directory subDir : getSubFolders(account, dirId)) {
+            subResults.addAll(searchFolder(account, subDir.getId(), query));
+        }
```

Comment:
This could be a stream.

Reply:
Maybe this is trickier than I thought because of the recursive exception.

Reply:
Do you also have a suggestion for the stream here?

Reply:
I tried my self, but maybe stream is not appropiate here because of the recursive exception. So for loop is fine by me.